frontend/src/pages/IstioConfigNew/GatewayForm/ListenerBuilder.tsx

```diff
@@ -27,12 +32,28 @@ export const isValidHostname = (hostname: string): boolean => {
   return hostname !== undefined && hostname.length > 0 && isK8sGatewayHostValid(hostname);
 };
 
+export const isValidTLS = (protocol: string, tls?: K8sGatewayTLS | null): boolean => {
+  const tlsRequired = protocolsCert.includes(protocol);
+  if (!tls || !tlsRequired) {
+    return true;
+  }
+
+  for (const cert of tls.certificateRefs) {
+    const certsValid = tlsRequired ? cert.name?.length > 0 : true;
+    if (!certsValid) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
 export const isValidPort = (port: string): boolean => {
   return port.length > 0 && !isNaN(Number(port)) && Number(port) >= MIN_PORT && Number(port) <= MAX_PORT;
 };
 
 export const isValidSelector = (selector: string): boolean => {
-  return selector.length === 0 || typeof addSelectorLabels(selector) !== 'undefined';
+  return selector.length === 0 || addSelectorLabels(selector)[0];
```

Comment:
```suggestion
  return addSelectorLabels(selector)[0];
```
Now that the label input is only enabled for `Selector` option, empty value should be invalid.